File: TSDW/Esercitazioni/itinere3/game_l/app/Http/Controllers/GameController.php
<?php

namespace App\Http\Controllers;

use App\Models\Game;
use Illuminate\Http\Request;
use App\Models\Player;

class GameController extends Controller
{
    /**
     * Display a listing of the resource.
     */
    public function index()
    {
        return view("games.index", ["games"=>Game::all()]);
    }

    /**
     * Show the form for creating a new resource.
     */
    public function create()
    {
        return view("games.create", ["players"=>Player::all()]);
    }

    /**
     * Store a newly created resource in storage.
     */
    public function store(Request $request)
    {
        Game::create($request->all());
        return redirect()->route('games.index');
    }
    /**
     * Display the specified resource.
     */
    public function show(Game $game)
    {
        return view('games.show', ['game'=>$game]);
    }
    /**
     * Show the form for editing the specified resource.
     */
    public function edit(Game $game)
    {
        return view('games.edit', ['game'=>$game, 'players'=>Player::all()]);
    }

    /**
     * Update the specified resource in storage.
     */
    public function update(Request $request, Game $game)
    {
        $game->update($request->all());
        return redirect()->route('games.index');
    }

    /**
     * Remove the specified resource from storage.
     */
    public function destroy(Game $game)
    {
        $game->delete();
        return redirect()->route('games.index');
    }
}

File: TSDW/Esercitazioni/itinere3/game_s/src/main/java/com/example/game_s/controllers/GameController.java
package com.example.game_s.controllers;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.ui.Model;
import com.example.game_s.models.Game;
import com.example.game_s.repositories.GameRepository;
import com.example.game_s.repositories.PlayerRepository;

@Controller
public class GameController {
    
    private final PlayerRepository repoPlayer;
    private final GameRepository repo;

    public GameController(PlayerRepository repoPlayer, GameRepository repo){
        this.repoPlayer = repoPlayer;
        this.repo = repo;
    }

    @GetMapping("/game/create")
    public String create(){
        return "games/create";
    }

    @GetMapping("/games")
    public String index(Model model){
        model.addAttribute("games", repo.findAll());
        return "games/index";
    }

    @PostMapping("/games/store")
    public String store(@ModelAttribute Game game){
        repo.save(game);
        return "redirect:/games";
    }

    @GetMapping("/games/show")
    public String show(Model model, Long id){
        model.addAttribute("game", repo.findById(id).orElse(null));
        model.addAttribute("player", repoPlayer.findAll());
        return "games/show";
    }

    @PostMapping("/games/update")
    public String update(@ModelAttribute Game game){
        repo.save(game);
        return "redirect:/games";
    }

    @PostMapping("/games/delete")
    public String delete(Long id){
        repo.deleteById(id);
        return "redirect:/games";
    }
}
